File: Javascript/number.js
/*
 *   Copyright 2022. Eduardo Programador
 *   www.eduardoprogramador.com
 *   consultoria@eduardoprogramador.com
 *
 *   All Rights Reserved.
 *
 * */


class Number {        

    constructor(first = '',second = '') {
        if(first.localeCompare('') == 0 && second.localeCompare('') == 0) {
            this.first = null;
            this.second = null;
        } else if(first.localeCompare('0') != 0 && second.localeCompare('') == 0) {
            this.number = first;
            this.second = null;            
        } else if(first.localeCompare('0') != 0 && second.localeCompare('0') != 0) {
            this.first = first;
            this.second = second;
        }
    }    

    static listNumberTypes() {
        return new Array("Binary,Hexadecimal,Octal");
    }

    toDecimal();
    toDecimal(number);

    add();
    add(numOnenumTwo);

    smallest();
    smallest(numOne, numTwo);

    cutZeros();
    cutZeros(number);

    fillWithZeros(count);
    fillWithZeros(count,number);

    subtract();
    subtract(numOne, numTwo);

    multiply();
    multiply(num, by);

    divide();
    divide(numOne,by);

    modulus();
    modulus(numOne,by);
}

class Binary extends Number {
    //code...
}

File: Ruby/Number.rb
#/*
# *   Copyright 2022. Eduardo Programador
# *   www.eduardoprogramador.com
# *   consultoria@eduardoprogramador.com
# *
# *   All Rights Reserved.
# *
# * */

class Number 

    public

    def self.listNumberTypes()
        list = ["Binary,Hexadecimal,Octal"]
        return list
    end

    protected

    attr_accessor :number, :first, :second    
    
    def initialize(*args)
        case args.size
            when 0
                @number = nil
                @first = nil
                @second = nil
            when 1
                @number = args[0]
                @first = nil
                @second = nil
            when 2
                @number = nil
                @first = args[0]
                @second = args[1]
            else 
                raise LowLevelException.new("Invalid Number of Arguments")
        end
    end        
    
    def toDecimal(*args)
    end
        
    def add(*args)
    end    

    def smallest(*args)
    end

    def cutZeros(*args)
    end    

    def fillWithZeros(*args)
    end    

    def subtract(*args)
    end    

    def multiply(*args)
    end    

    def divide(*args)
    end    

    def modulus(*args)
    end    
end